lit/docs/operation/creds/kubernetes.lit

```diff
@@ -0,0 +1,251 @@
+  ---
+  # Identifies the `web` service as an actor.
+  #
+  # ref: https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/
+  #
+  apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    name: web
+    labels:
+      app: web
+
+
+  ---
+  # Describes the rules that represents the set of permissions
+  # that an actor can perform in any namespace within the cluster.
+  #
+  # ref: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#api-overview
+  #
+  apiVersion: rbac.authorization.k8s.io/v1beta1
+  kind: ClusterRole
+  metadata:
+    name: read-secrets
+    labels:
+      app: web
+  rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get"]
+
+
+  ---
+  # Grants the permissions defined in a role to one or more 
+  # users (subjects) represented as `ServiceAccount`s.
+  #
+  # Even though in this example we're binding to a ClusterRole
+  # (which is not tied to any namespace), the use of such cluster
+  # role is (see `metadata.namespace`), making the effective 
+  # permissions restricted to the namespace applied in the
+  # `roleBinding`.
+  #
+  # To have access to multiple namespaces, multiple RoleBindings
+  # would then be created (one for each namespace).
+  #
+  # ref: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding
+  #
+  apiVersion: rbac.authorization.k8s.io/v1beta1
+  kind: RoleBinding
+  metadata:
+    name: web-team-a
+    namespace: myprefix-team-a
+    labels:
+      app: web
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: ClusterRole
+    name: read-secrets
+  subjects:
+  - kind: ServiceAccount
+    name: web
+    namespace: concourse
+
+  ---
+  # Role binding for the second team (`team-b`), allowing `web`
+  # to consume secrets from it.
+  #
+  apiVersion: rbac.authorization.k8s.io/v1beta1
+  kind: RoleBinding
+  metadata:
+    name: web-team-b
+    namespace: myprefix-team-b
+    labels:
+      app: web
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: ClusterRole
+    name: read-secrets
+  subjects:
+  - kind: ServiceAccount
+    name: web
+    namespace: concourse
+
+  ---
+  # The deployment that creates the `web` pods.
+  # Note how it ties the pods to the `ServiceAccount`
+  # previously created.
+  #
+  # ref: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/
+  # ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/
+  #
+  apiVersion: extensions/v1beta1
+  kind: Deployment
+  metadata:
+    name: web
+    labels:
+      app: web
+  spec:
+    replicas: 1
+    template:
+      metadata:
+        labels:
+          app: web
+      spec:
+        serviceAccountName: web
+        containers:
+          - name: web
+            image: "concourse/concourse:5.2.0"
+            args: [ web ]
+            env:
+              - name: CONCOURSE_KUBERNETES_NAMESPACE_PREFIX
+                value: "myprefix-"
+            # ...
+  }}}
```

Comment:
this YAML file is quite big 😬 I'm not sure if this would be the best way of presenting it - I tried separating them into separate code blocks, but I feel that this way looks a bit better (and it's quite common to see docs like this when it comes to k8s).

wdyt?

Reply:
![Screen Shot 2019-05-30 at 15 15 28-fullpage](https://user-images.githubusercontent.com/3574444/58658111-c70eb280-82ed-11e9-8de3-dc90653769f8.png)

Reply:
Would an alternative be to split it up and move the comments into the Booklit document? Or is having it all in one YAML document useful for copy-pasting and submitting everything at once?

Reply:
ooooh good point, those side comments would be nice. I'll give them a try here 👍  


> Or is having it all in one YAML document useful for copy-pasting and submitting everything at once?

hmmm I saw a couple of times people having multiple yaml blocks altogether when it comes to k8s, but I'd personally split 😅

Reply:
after splitting, seems easier to follow; wdyt?

![Screen Shot 2019-06-20 at 16 34 44-fullpage](https://user-images.githubusercontent.com/3574444/59879833-a16b4b00-9379-11e9-8382-00624be3dded.png)